src/event_handler/code_expantion.rs

```diff
@@ -0,0 +1,109 @@
+use color_eyre::eyre::Result;
+use poise::serenity_prelude::{self as serenity, Context, CreateMessage, FullEvent};
+use regex::Regex;
+use reqwest::Client;
+
+pub async fn handle(ctx: &Context, event: &FullEvent) -> Result<()> {
+    let pattern =
+        r"https?:\/\/github\.com\/([\w-]+\/[\w.-]+)\/blob\/(.+?)\/(.+?)#L(\d+)[~-]?L?(\d*)";
+    let re = Regex::new(pattern)?;
+
+    if let FullEvent::Message { new_message } = event {
+        let content = &new_message.content;
+        if re.captures(content.as_str()).is_some() {
+            let mut code_blocks = Vec::new();
+
+            for caps in re.captures_iter(content) {
+                let full_url = caps.get(0).unwrap().as_str();
+                let repo = caps.get(1).unwrap().as_str();
+                let reference = caps.get(2).unwrap().as_str();
+                let file = caps.get(3).unwrap().as_str();
+                let start_str = caps.get(4).unwrap().as_str();
+                let start = start_str.parse::<usize>().unwrap_or(1);
+                let end = if let Some(end_str) = caps.get(5) {
+                    end_str.as_str().parse::<usize>().unwrap_or(start)
+                } else {
+                    start
+                };
```

Comment:
this is cursed

Reply:
tell me about it

Reply:
working on some chef work, cursed regex magic for gitea/forgejo instance hopefully looks a bit cleaner